docs/2.guide/2.directory-structure/1.shared.md

```diff
@@ -0,0 +1,92 @@
+﻿---
+title: 'shared'
+head.title: 'shared/'
+description: Use the shared/ directory to share functionality between the Vue app and the Nitro server.
+navigation.icon: i-ph-folder
+---
+
+The `shared/` directory allows you to share code that can be used in both the Vue app and the Nitro server.
+
+::tip
+The `shared/` directory is available in Nuxt v3.14+.
+::
+
+::important
+Code in the shared directory cannot import any Vue or nitro code.
+::
+
+## Usage
+
+**Method 1:** Using named export
+
+```ts twoslash [shared/utils/capitalize.ts]
+export const capitalize = (input: string) => {
+  return input[0].toUpperCase() + input.slice(1)
+}
+```
+
+**Method 2:** Using default export
+
+```ts twoslash [shared/utils/capitalize.ts]
+export default function capitalize (input: string) {
+  return input[0].toUpperCase() + input.slice(1)
+}
+```
+
+**Usage:** You can now use auto-imported utility functions in `.js`, `.ts` and `.vue` files within your Vue app and the `server/` directory. 
+
+If you have set `compatibilityVersion: 4` in your `nuxt.config.ts`, you can use the auto-imported functions in the `app/` directory. This is part of Nuxt's progressive compatibility features preparing for version 4.
```

Comment:
How about adding a link to [Opting in to Nuxt 4](https://nuxt.com/docs/getting-started/upgrade#opting-in-to-nuxt-4)?

For example:
```
If you have [set `compatibilityVersion: 4` in your `nuxt.config.ts`](https://nuxt.com/docs/getting-started/upgrade#opting-in-to-nuxt-4), you can use the auto-imported functions in the `app/` directory. This is part of Nuxt's progressive compatibility features preparing for version 4.
```

Reply:
That's a good shout, I've added a link to the opt in page.